source/MaterialXGenShader/Shader.h

```diff
@@ -135,15 +153,15 @@ class Shader
 
     /// Create a new constant variable for a stage.
     virtual void createConstant(size_t stage, const TypeDesc* type, const string& name,
-        const string& semantic = EMPTY_STRING, ValuePtr value = nullptr);
+                                const string& elementPath, const string& semantic = EMPTY_STRING, ValuePtr value = nullptr);
 
     /// Create a new variable block for uniform inputs in a stage.
     virtual void createUniformBlock(size_t stage, const string& block, const string& instance = EMPTY_STRING);
 
     /// Create a new variable for uniform data in the given block for a stage.
     /// The block must be previously created with createUniformBlock.
     virtual void createUniform(size_t stage, const string& block, const TypeDesc* type, const string& name,
-        const string& semantic = EMPTY_STRING, ValuePtr value = nullptr);
+                               const string& elementPath, const string& semantic = EMPTY_STRING, ValuePtr value = nullptr);
```

Comment:
Could you set a default value of EMPTY_STRING to the elementPath here, to avoid all the EMPTY_STRING additions in the code? 
Do the same on all methods that create shader variables.

Reply:
Yes, Will do.